julia-mode.el

```diff
@@ -908,6 +919,40 @@ buffer where the LaTeX symbol starts."
           (abbrev-insert symb name beg end)))
     #'ignore))
 
+(defun julia-mode--latexsub-before-point ()
+  "When there is a LaTeX substitution that can be made before the point, return (CONS BEG LATEX).
+
+`beg' is the position of the `\`, `latex' is the string to replace, including the `\`.
+
+When multiple options match, ask the user to clarify via `completing-read', unless there is a complete match and `julia-latexsub-greedy' is `t'."
+  (when-let (beg (julia--latexsub-start-symbol))
+    (let ((partial (buffer-substring-no-properties beg (point))))
+      (message "partial %s" partial)
+      (when-let (replacements (gethash partial julia-mode--latexsubs-partials))
+        (message "replacements %s" (prin1 replacements))
+        (let* ((complete-match (member partial replacements))
+               (replacement (cond ((and complete-match julia-latexsub-greedy) partial)
+                                  ((cdr replacements) (gethash (completing-read "LaTeX completions: " replacements) julia-mode-latexsubs))
```

Comment:
I think this is an error.

When the completion is not full, e.g., \cir<TAB>, I am offered a list of valid completions. When I press enter on one of them, I get a `progn: Wrong type argument: char-or-string-p, nil` error.

After some inspection, it seems that this is because this line here should return the raw latex control sequence, not the already substituted value. The substitution is performed in the  `julia-latexsub-or-indent` function.

If I change this line to

```elisp
((cdr replacements) (completing-read "LaTeX completions: " replacements))
```

then it works as expected.